src/Avalonia.Controls/ItemsSourceView.cs

```diff
@@ -82,25 +90,60 @@ public ItemsSourceView(IEnumerable source)
         /// <summary>
         /// Occurs when the collection has changed to indicate the reason for the change and which items changed.
         /// </summary>
-        public event NotifyCollectionChangedEventHandler? CollectionChanged;
+        public event NotifyCollectionChangedEventHandler? CollectionChanged
+        {
+            add
+            {
+                if (_inner is null)
+                    ThrowDisposed();
```

Comment:
Can we reference ```Inner``` below instead of that check?

Reply:
Good points! Fixed.